test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,242 @@
+
+void fastq_read_from_disk_seqan3(benchmark::State & state)
+{
+    auto file = create_fastq_file(state.range(0));
+    auto path = file.get_path();
+
+    std::vector<seqan3::dna5_vector> sequences{};
+    std::vector<std::string> ids{};
+    std::vector<std::vector<seqan3::phred42>> qualities{};
+
+    for (auto _ : state)
+    {
+        seqan3::sequence_file_input fin{path};
+
+        for (auto && [read_sequence, read_id, read_quality] : fin)
+        {
+            sequences.push_back(std::move(read_sequence));
+            ids.push_back(std::move(read_id));
+            qualities.push_back(std::move(read_quality));
+        }
+
+        sequences.clear();
+        ids.clear();
+        qualities.clear();
+    }
+}
+
+// ============================================================================
+// seqan2 benchmark
+// ============================================================================
+
+#if SEQAN3_HAS_SEQAN2
+
+// ----------------------------------------------------------------------------
+// read dummy fastq file from a stream
+// ----------------------------------------------------------------------------
+
+void fastq_read_from_stream_seqan2(benchmark::State & state)
+{
+    std::istringstream istream{};
+    size_t entries_size = state.range(0);
+    auto file = generate_fastq_file(entries_size);
+
+    auto restart_iterator = [&istream, &file]()    // cf. format_fasta_benchmark
+    {
+        istream = std::istringstream{file};
+        // same constant as seqan3 benchmark
+
+        seqan::VirtualStream<char, seqan::Input> comp{};
+        seqan::open(comp, istream);
+        return seqan::directionIterator(comp, seqan::Input());
+    };
+
+    seqan::String<char> id{};
+    seqan::String<seqan::Dna5> seq{};
+    seqan::String<char> qual{};
+
+    for (auto _ : state)
+    {
+        state.PauseTiming();
+        auto it = restart_iterator();
+        state.ResumeTiming();
+
+        for (size_t i = 0; i < entries_size; ++i)
+            seqan::readRecord(id, seq, qual, it, seqan::Fastq{});
+
+        seqan::clear(id);
+        seqan::clear(seq);
+        seqan::clear(qual);
+    }
+}
+
+// ----------------------------------------------------------------------------
+// read dummy fastq file from temporary file on disk
+// ----------------------------------------------------------------------------
+
+void fastq_read_from_disk_seqan2(benchmark::State & state)
+{
+    seqan3::test::tmp_filename file_name{"tmp.fastq"};
+    auto tmp_path = file_name.get_path();
+
+    std::ofstream ostream{tmp_path};
+    ostream << generate_fastq_file(state.range(0));
+    ostream.close();
+
+    seqan::StringSet<seqan::String<char>> ids{};
+    seqan::StringSet<seqan::String<seqan::Dna5>> seqs{};
+    seqan::StringSet<seqan::String<char>> quals{};
+
+    for (auto _ : state)
+    {
+        seqan::SeqFileIn seqFileIn(tmp_path.c_str());
+        seqan::readRecords(ids, seqs, quals, seqFileIn);
```

Comment:
Why does the reading-from-stream-benhcmark is done with single `readRecord` calls and reading-from-disc is done with chunk-reading  `readRecords`? Is this a fair comparison compared to reading line by line with seqan3?

Reply:
Dunno

Reply:
The messaage was: `the PR is good as is just needs some more love`. But now everything has changed anyway. So I adapt this as well.